### DISCUSSION TITLE
235

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#748] [Comments]
http://jojoldu.tistory.com/235

**Comment by jojoldu** [comment]
작성자: 이호진

제가 이글을 이글을 확인할수 있을지는 모르겠지만
deleteInBatch 를 사용하도 같은결과 인가요?

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

직접 테스트해보진 않았는데요
말씀주신것보고 찾아보니 delete in batch는 deleteAll과 달리 건별로 삭제가 아닌 전체 삭제 쿼리로 생성되서 진행되는것 같습니다!

https://stackoverflow.com/questions/26142261/spring-jparepostory-delete-vs-deleteinbatch

**Comment by jojoldu** [comment]
작성자: 이호진

저번에 deleteInBatch로 삭제한적이 있어서 혹시나하고 말씀드린거에요

블로그 내용들이 다 좋네요
좋은 구경 하고 가겠습니다

감사합니다

**Comment by jojoldu** [comment]
작성자: 창천향로

크으! 감사합니다!! 좋은정보 알게됐습니다!

**Comment by cheese10yun** [comment]
좋은글 감사합니다

**Comment by pjh5512** [comment]
좋은 글입니다. JPA 공부중인데 많은 도움이 되었습니다.

**Comment by nswon** [comment]
안녕하세요! 궁금한 점이 있어서 질문드립니다. 
저도 벌크연산을 사용해서 대량의 id를 삭제시키고 있는데, 
만약 존재하지 않는 id일 경우 커스텀 예외를 터트리고 싶어요! 이럴때는 벌크연산 말고 다른 방법을 사용해야 할 것 같은데 어떤 방법이 있을지 궁금해요.

**Reply by Juhyeok0202** [reply]
저도 이 부분이 궁금해서 고민 중인데 혹시 어떤 결론이 도출되셨나요??
